contracts/interfaces/V3SpokePoolInterface.sol

```diff
@@ -97,67 +97,82 @@ interface V3SpokePoolInterface {
      **************************************/
 
     event V3FundsDeposited(
-        address inputToken,
-        address outputToken,
+        bytes32 inputToken,
+        bytes32 outputToken,
         uint256 inputAmount,
         uint256 outputAmount,
         uint256 indexed destinationChainId,
         uint32 indexed depositId,
         uint32 quoteTimestamp,
         uint32 fillDeadline,
         uint32 exclusivityDeadline,
-        address indexed depositor,
-        address recipient,
-        address exclusiveRelayer,
+        bytes32 indexed depositor,
+        bytes32 recipient,
+        bytes32 exclusiveRelayer,
         bytes message
     );
 
     event RequestedSpeedUpV3Deposit(
         uint256 updatedOutputAmount,
         uint32 indexed depositId,
-        address indexed depositor,
-        address updatedRecipient,
+        bytes32 indexed depositor,
+        bytes32 updatedRecipient,
         bytes updatedMessage,
         bytes depositorSignature
     );
 
     event FilledV3Relay(
-        address inputToken,
-        address outputToken,
+        bytes32 inputToken,
+        bytes32 outputToken,
         uint256 inputAmount,
         uint256 outputAmount,
         uint256 repaymentChainId,
         uint256 indexed originChainId,
         uint32 indexed depositId,
         uint32 fillDeadline,
         uint32 exclusivityDeadline,
-        address exclusiveRelayer,
-        address indexed relayer,
-        address depositor,
-        address recipient,
+        bytes32 exclusiveRelayer,
+        bytes32 indexed relayer,
+        bytes32 depositor,
+        bytes32 recipient,
         bytes message,
         V3RelayExecutionEventInfo relayExecutionInfo
     );
 
     event RequestedV3SlowFill(
-        address inputToken,
-        address outputToken,
+        bytes32 inputToken,
+        bytes32 outputToken,
         uint256 inputAmount,
         uint256 outputAmount,
         uint256 indexed originChainId,
         uint32 indexed depositId,
         uint32 fillDeadline,
         uint32 exclusivityDeadline,
-        address exclusiveRelayer,
-        address depositor,
-        address recipient,
+        bytes32 exclusiveRelayer,
+        bytes32 depositor,
+        bytes32 recipient,
         bytes message
     );
 
     /**************************************
      *              FUNCTIONS             *
      **************************************/
 
+    function depositV3(
+        bytes32 depositor,
+        bytes32 recipient,
+        bytes32 inputToken,
+        bytes32 outputToken,
+        uint256 inputAmount,
+        uint256 outputAmount,
+        uint256 destinationChainId,
+        bytes32 exclusiveRelayer,
+        uint32 quoteTimestamp,
+        uint32 fillDeadline,
+        uint32 exclusivityDeadline,
+        bytes calldata message
+    ) external payable;
+
     function depositV3(
         address depositor,
```

Comment:
Do we need to support backwards overload in the interface?

Reply:
Good point. I've removed them from the interface 👍